polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx

```diff
@@ -198,20 +197,42 @@ export class PositionedOverlay extends PureComponent<
     this.overlay = node;
   };
 
```

Comment:
Currently this is built to add vertically scrolling containers and not horizontally scrolling containers, but horizontal scroll support can be easily added if needed.

Performance is actually better than before since `Scrollable.forNode` was repeating a DOM search for each parentNode from the activator. We are now doing single traversal.

Reply:
Ah interesting. One thing I'm seeing is that if there are enough rows in the bulk editor to cause the page to be vertically scrollable, then the popovers are also tracking with horizontal scrolling (I'm guessing since it is the same container that scrolls). But without a vertical scrollbar, the popovers do not track horizontally.

e.g. 
- [No vertical scrollbar:](https://admin.web.web-ya2s.sophie-schneider.us.spin.dev/store/shop1/bulk?resource_name=Product&edit=description%2Cmedia%2Ctags%2Cstatus%2Cproduct_taxonomy_node_id%2Cproduct_type%2Cvendor%2Ctemplate_suffix%2Cpublications.2.published_at%2Cvariants.price%2Cvariants.inventory_on_hand_adjustment&return_to=%2Fstore%2Fshop1%2Fproducts%3FselectedView%3Dall&selectedView=all&order=title+ascending&ids=22%2C8)
- [With vertical scrollbar](https://admin.web.web-ya2s.sophie-schneider.us.spin.dev/store/shop1/bulk?resource_name=Product&edit=description%2Cmedia%2Ctags%2Cstatus%2Cproduct_taxonomy_node_id%2Cproduct_type%2Cvendor%2Ctemplate_suffix%2Cpublications.2.published_at%2Cvariants.price%2Cvariants.inventory_on_hand_adjustment&return_to=%2Fstore%2Fshop1%2Fproducts%3FselectedView%3Dall&selectedView=all&order=title+ascending&ids=22%2C8%2C2%2C17%2C25%2C5%2C29%2C26%2C20%2C4%2C28%2C13%2C18%2C6%2C31%2C9%2C11%2C30%2C1%2C23%2C27%2C15%2C21%2C19%2C24%2C7%2C14%2C10%2C16%2C12%2C3)
 

Thoughts on adding horizontal containers as well?

Reply:
Ah it looks like the `PositionedOverlay` doesn't calculate [horizontal position](https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/PositionedOverlay/utilities/math.ts#L87-L92) from the scroll containers like it does with the [vertical position](https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/PositionedOverlay/utilities/math.ts#L17). I'm going to have to add that logic for horizontal scrolls too

Reply:
@marty-Wallace good news! I got it to work for horizontal scroll for both your examples above, but theres a problem with the bulk editor where the [scroll container is registered as 100% full screen width](https://github.com/Shopify/web/blob/main/app/shared/components/BulkEditor/components/Table/Table.scss#L9) behind the sticky column so it doesn't close the popover until the end of that scroll container. I can't really think of a way to fix this from polaris' end so we might have to modify the width of the bulk editor scroll container or make the sticky column z-index higher than the popover

https://github.com/Shopify/polaris/assets/20652326/6b6aed25-9d59-41d2-a7ed-f1c5753af735

Reply:
I think we should involve @e7chun and @kristy-chen2023 in the discussion as how we should proceed seems more like a product/UX decision.

Reply:
@e7chun they can be updated at the same time but not in this PR, it would be when the next polaris version is bumped in web that includes this change

Reply:
Since the popover staying visible when the activator is behind the sticky columns is already a bug in production, we shouldn't block this PR from shipping now. @sophschneider

Reply:
@chloerice do we need it for your HoverCard PR? I was holding off on it in case there were conflicts and wanted you to merge in first